src/integration-test/java/com/commercetools/sync/integration/producttypes/ProductTypeSyncIT.java

```diff
@@ -0,0 +1,827 @@
+import static com.commercetools.sync.integration.producttypes.utils.ProductTypeITUtils.populateSourceProject;
+import static com.commercetools.sync.integration.producttypes.utils.ProductTypeITUtils.populateTargetProject;
+import static java.util.Collections.singletonList;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ProductTypeSyncIT {
+
+    /**
+     * Deletes product types from source and target CTP projects.
+     * Populates source and target CTP projects with test data.
+     */
+    @Before
+    public void setup() {
+        deleteProductTypesFromTargetAndSource();
+        populateSourceProject();
+        populateTargetProject();
+    }
+
+    /**
+     * Deletes all the test data from the {@code CTP_SOURCE_CLIENT} and the {@code CTP_SOURCE_CLIENT} projects that
+     * were set up in this test class.
+     */
+    @AfterClass
+    public static void tearDown() {
+        deleteProductTypesFromTargetAndSource();
+    }
+
+    @Test
+    public void sync_WithUpdatedProductType_ShouldUpdateProductType() {
+        final Optional<ProductType> oldProductTypeBefore = getProductTypeByKey(CTP_TARGET_CLIENT, PRODUCT_TYPE_KEY_1);
+        assertThat(oldProductTypeBefore).isNotEmpty();
+
+
+        final ProductTypeDraft newProductTypeDraft = ProductTypeDraft.ofAttributeDefinitionDrafts(
+            PRODUCT_TYPE_KEY_1,
+            PRODUCT_TYPE_NAME_2,
+            PRODUCT_TYPE_DESCRIPTION_2,
+            singletonList(ATTRIBUTE_DEFINITION_DRAFT_1)
+        );
+
+        final ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder
+            .of(CTP_TARGET_CLIENT)
+            .build();
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(productTypeSyncOptions);
+
+        final ProductTypeSyncStatistics productTypeSyncStatistics = productTypeSync
+            .sync(singletonList(newProductTypeDraft))
+            .toCompletableFuture().join();
+
+        AssertionsForStatistics.assertThat(productTypeSyncStatistics).hasValues(1, 0, 1, 0);
+
+
+        final Optional<ProductType> oldProductTypeAfter = getProductTypeByKey(CTP_TARGET_CLIENT, PRODUCT_TYPE_KEY_1);
+
+        assertThat(oldProductTypeAfter).isNotEmpty();
+        assertThat(oldProductTypeAfter.get().getName()).isEqualTo(PRODUCT_TYPE_NAME_2);
+        assertThat(oldProductTypeAfter.get().getDescription()).isEqualTo(PRODUCT_TYPE_DESCRIPTION_2);
+        assertAttributesAreEqual(oldProductTypeAfter.get().getAttributes(),
+            singletonList(ATTRIBUTE_DEFINITION_DRAFT_1));
```

Comment:
instead of those 4 asserts and checking that it is not empty explicitly and doing `.get()` after. You can do it like this:

````java
assertThat(oldProductTypeAfter).hasValueSatisfying(productType -> {
            assertThat(productType.getName()).isEqualTo(PRODUCT_TYPE_NAME_2);
            assertThat(productType.getDescription()).isEqualTo(PRODUCT_TYPE_DESCRIPTION_2);
            assertAttributesAreEqual(productType.getAttributes(), singletonList(ATTRIBUTE_DEFINITION_DRAFT_1));
        });
````

Reply:
you can use it across all asserts of this class actually ;)

Reply:
601ffb1e1af54363e7d5a52b771054f5d7d1a900